### PR TITLE
various changes for removing warning in clang compilation: const char…

### PR DESCRIPTION
**Pull request by eruffaldi**
Fix of building error under macOS clang 9.1.0 and warnings under Linux with GCC 4.9.2

### CODE REVIEW

File: ndicapi.cxx [hunk 1]
```diff
@@ -68,7 +68,7 @@ namespace
     // call the user-supplied callback function
     if (pol->ErrorCallback)
     {
-      pol->ErrorCallback(errnum, ndiErrorString(errnum), pol->ErrorCallbackData);
+      pol->ErrorCallback(errnum, (char*)ndiErrorString(errnum), pol->ErrorCallbackData);
```
Comment (adamrankin) [inline ndicapi.cxx:73]: So, the return type of ndiErrorString is `char*`, so this is redundant. Is it the `ndicapiExport` token in `ndicapiExport char* ndiErrorString(int errnum);` that causes the warning?

Comment (adamrankin) [reply ndicapi.cxx:73]: Oh, I see, changes to ndiErrorString below, nevermind.

File: ndicapimodule.cxx [hunk 2]
```diff
@@ -25,6 +25,7 @@
   #define PyString_Format PyUnicode_Format
   #define PyString_AsString PyUnicode_AsUTF8
   #define PyIntObject PyLongObject
+  //#define PY_INT_OBJECT_OB_IVAL(ob) PyLong_AsLong((PyObject*)(ob))
```
Comment (adamrankin) [inline ndicapimodule.cxx:28]: Please remove commented code. If you want to keep it, please add a comment above as to why it's being kept.

File: ndicapimodule.cxx [hunk 3]
```diff
@@ -185,7 +186,7 @@ bitfield_invert(PyIntObject* v)
 static PyObject*
 bitfield_lshift(PyIntObject* v, PyIntObject* w)
 {
-  register unsigned long a, b;
+  register long a, b;
```
Comment (adamrankin) [inline ndicapimodule.cxx:189]: Is this a bug fix or a compiler warning? If compiler warning, please `#ifdef` it to OSX vs other

File: ndicapimodule.cxx [hunk 4]
```diff
@@ -1527,9 +1528,9 @@ ndicapiExport MOD_INIT(ndicapy)
 
   Py_NDIErrcodeMacro(NDI_ENVIRONMENT);
 
-  Py_NDIErrcodeMacro(NDI_EPROM_READ);
-  Py_NDIErrcodeMacro(NDI_EPROM_WRITE);
-  Py_NDIErrcodeMacro(NDI_EPROM_ERASE);
+  //Py_NDIErrcodeMacro(NDI_EPROM_READ);
```
Comment (adamrankin) [inline ndicapimodule.cxx:1623]: Do these no longer work? If so, could you add an issue?

If they do work, please undo the commenting.

**Comment by adamrankin** [comment]
Thank you very much for the fixes! I've added some comments and questions to the PR.

File: ndicapimodule.cxx (hunk 4 above)
Comment (eruffaldi) [reply ndicapimodule.cxx:1623]: Removed due to missing symblols. I can protect them with an #ifdef

File: ndicapimodule.cxx (hunk 3 above)
Comment (eruffaldi) [reply ndicapimodule.cxx:189]: Compiler mega-warning when doing (b < 0) that's impossible. These variables got assigned as from ob_digit[0] from PyLong that is a signed long and not unsigned. 

I would also remove the register keyword.

File: ndicapimodule.cxx (hunk 2 above)
Comment (eruffaldi) [reply ndicapimodule.cxx:28]: The safe way to get the long inside a PyLong is to use PyLong_AsLong(PyObject*) and not just ob_digit[0]. All around in CPython PyLong_AsLong is used. 
https://github.com/python/cpython/blob/6405feecda6a5d5dd7a4240eb3054a2676ed29b1/Objects/longobject.c

Comment (adamrankin) [reply ndicapimodule.cxx:28]: I think you know more about this than I do. Whatever you think is right (and works on all platforms).

**Comment by adamrankin** [comment]
After a couple of `#ifdef`s, let me know and I'll merge it.

File: ndicapimodule.cxx (hunk 2 above)
Comment (eruffaldi) [reply ndicapimodule.cxx:28]: Fixed with the most compatible way: PyLong_AsLong

**Comment by adamrankin** [comment]
Thanks for the contribution, the python wrapping is not my area!

**Pull request merged**